### PR TITLE
Add circle ci changes

### PR DESCRIPTION
**Pull request by linonymous**
Signed-off-by: linonymous <iamlinonymous@gmail.com>

### CODE REVIEW

**Review by ajinkyan83 — APPROVED** [review]
/lgtm

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - approve-prod:
+          type: approval
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+(\.\d+)?(\.\d+)?$/
+          requires:
+            - terraform-plan
+      - terraform-apply:
```
Comment (fayazg) [inline .circleci/config.yml:85]: shouldn't this be `deploy_prod`?

**Review by fayazg — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    environment:
+      AWS_ACCESS_KEY_ID_ENV_VAR: AWS_ACCESS_KEY_ID_TEST
+      AWS_SECRET_ACCESS_KEY_ENV_VAR: AWS_SECRET_ACCESS_KEY_TEST
+      AWS_REGION_ENV: AWS_REGION_TEST
+      STAGE: test
+  deploy_prod:
+    <<: *deploy
+    filters:
+      tags:
+        only: /.*/
```
Comment (fayazg) [inline .circleci/config.yml:55]: this is part of the workflow jobs at the bottom so not needed here

**Review by fayazg — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (linonymous) [reply .circleci/config.yml]: done!

File: .circleci/config.yml (hunk 2 above)
Comment (linonymous) [reply .circleci/config.yml]: yes.!

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      STAGE: prod
+workflows:
+  version: 2
+  plan_approve_apply:
+    jobs:
+      - terraform-plan
+      - deploy_dev:
+          filters:
+            branches:
+              only: main
```
Comment (fayazg) [inline .circleci/config.yml:66]: doesn't this require terraform-plan?

Comment (linonymous) [reply .circleci/config.yml:66]: added.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      - deploy_test:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+(\.\d+)?(\.\d+)?$/
```
[the remark this answers is not in the document]
Comment (linonymous) [reply .circleci/config.yml:74]: added.

File: deploy/main.tf [hunk 5]
```diff
@@ -25,8 +25,8 @@ resource "aws_ecs_task_definition" "insights-git-task" {
   task_role_arn = aws_iam_role.ecs_task_role.arn
   container_definitions = jsonencode([
     {
-      name      = "insights-git"
-      image     = "linonymous/insights-git-binary:latest"
+      name      = "insights-connector-git"
+      image     = "395594542180.dkr.ecr.us-east-1.amazonaws.com/insights-git-connector:latest"
```
Comment (ajinkyan83) [inline deploy/main.tf:29]: @linonymous image name should be `insights-connector-git` as per convention.
CC: @fayazg

Comment (linonymous) [reply deploy/main.tf]: done.

**Pull request merged**